test/Client.test.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const express = require('express');
+const httpProxy = require('http-proxy-middleware');
+const request = require('supertest');
+const addEntries = require('../lib/utils/addEntries');
+const helper = require('./helper');
+const config = require('./fixtures/client-config/webpack.config');
+const runBrowser = require('./helpers/run-browser');
+
+function startProxy(port) {
+  const proxy = express();
+  proxy.use(
+    '/',
+    httpProxy({
+      target: 'http://localhost:9001',
+      ws: true,
+      changeOrigin: true,
+    })
+  );
+  return proxy.listen(port);
+}
+
+describe('Client code', () => {
+  beforeAll((done) => {
+    const options = {
+      compress: true,
+      port: 9001,
+      host: '0.0.0.0',
+      disableHostCheck: true,
+      hot: true,
+      watchOptions: {
+        poll: true,
+      },
+    };
+    addEntries(config, options);
```

Comment:
/cc @rlamana Looks we catch bug, when we use `API`, hot plugin should be added too, can you create issue?

Reply:
I assumed that was the API expected behavior, so I didn't think it was a bug. It all makes sense now. I will create the issue then.

Reply:
I created issue #1703.